multiqc/modules/ivar/ivar.py

```diff
@@ -0,0 +1,167 @@
+        
+    # Parse an ivar report
+    def parse_ivar(self, f):
+        parsed_data = dict()
+        regexes = {
+            'mapped_reads': r'(?:Found\s)(\d+)(?:\smapped)',
+            'total_reads': r'(?:Trimmed primers from )(?:\d+\.\d+\% \()?(\d+)',
+            'reads_outside_primer_region': r'^(?:\d+\.\d+\% \()?(\d+)(?:\))?(?:.*[of]?)reads\s(?:that\s)?started',
+            'reads_too_short_after_trimming': r'^(?:\d+\.\d+\% \()?(\d+)(?:\))?(?:.*[of]?)reads\swere(?: quality trimmed | shortened)'
+        }
+        for l in f['f']:
+            # Search regexes for stats
+            for k, r in regexes.items():
+                match = re.search(r, l)
+                if match:
+                    parsed_data[k] = int(match.group(1))
+        
+        return parsed_data
+
+    #Parse Primer stats appropriately
+    def parse_ivar_primer_stats(self, f):
+        primers = OrderedDict()
+        regex = "^(.*)(?:\t+)(\d+$)"
+        # Search regexes for stats
+        for l in f['f'].splitlines():
+            match = re.search(regex, l)
+            if match:
+                primer = match.group(1)
+                counts = int(match.group(2))
+                primers[primer] = counts
+        log.info("Found {} primers".format(len(primers)))
+        return primers
+
+    # Add to general stats table
+
+    def ivar_general_stats_table(self):
+        """ Take the parsed stats from the iVAR report and add it to the
+        basic stats table"""
+
+        headers = OrderedDict()
+        headers['mapped_reads'] = {
+            'title': 'Total mapped reads',
+            'description': 'Total number of mapped reads in iVar input.',
+            'min': 0,
+            'scale': 'RdYlGn-rev',
+            'format': '{:,.0f}'
+        }
+        headers['total_reads'] = {
+            'title': 'Primer trimmed reads',
+            'description': 'Total number of reads where primer trimming was performed.',
+            'min': 0,
+            'scale': 'PuRd',
+            'format': '{:,.0f}'
+        }
+        headers['reads_too_short_after_trimming'] = {
+            'title': 'Fail minimum length reads',
+            'description': 'Number of reads too short (<30bp) after primer trimming',
+            'min': 0,
+            'scale': 'RdYlGn',
+            'format': '{:,.0f}'
+        }
+        headers['reads_outside_primer_region'] = {
+            'title': 'Reads outside primer region',
+            'description': 'Number of reads outside the primer region',
+            'scale': 'RdYlGn-rev',
+            'format': '{:,.0f}'
+        }
+        self.general_stats_addcols(self.ivar_data, headers)
+    
+
+    def status_heatmap(self):
+        """ Heatmap showing information on each primer found for every sample """
+        data = self.parsed_primers
+        #print(data)
+        #Top level dict contains sample IDs + OrderedDict(primer, counts)
+        
+        final_data = list()
+        final_xcats = list()
```

Comment:
```suggestion
        final_xcats = set()
```
If you use a `set` instead of a `list` then the values must be unique (like a `factor` in R), so then you can just update it freely.

Reply:
Ah, quite possibly. Maybe better to leave it as it is if it works 👍

Reply:
Yeah, looks like sets are unordered at least in Python 2x. Lists are probably better then, sorry.. 🙏